components/usage/pkg/db/cost_center.go

```diff
@@ -233,3 +233,31 @@ func (c *CostCenterManager) ComputeInvoiceUsageRecord(ctx context.Context, attri
 		Draft:         false,
 	}, nil
 }
+
+func (c *CostCenterManager) ListLatestCostCentersWithBillingTimeBefore(ctx context.Context, strategy BillingStrategy, billingTimeBefore time.Time) ([]CostCenter, error) {
+	db := c.conn.WithContext(ctx).Debug()
+
+	var results []CostCenter
+	var batch []CostCenter
+
+	subquery := db.
+		Table((&CostCenter{}).TableName()).
+		Select("DISTINCT id, MAX(creationTime) AS creationTime").
+		Where("billingStrategy = ?", strategy).
+		Group("id").
+		Where("nextBillingTime != ?", "").
+		Where("nextBillingTime < ?", TimeToISO8601(billingTimeBefore))
```

Comment:
All conditions need to be applied to the main query, otherwise we wouldn't necessarily get the latest cost center per attributionid.

Reply:
We do the join `ON (id, creationTime)` so we should get it. The unit test in this PR also looks to work.

Reply:
I think the subselect doesn't fetch the latest version of each cost center, but the latest one fulfilling the conditions. So if say I change to `stripe` it would pick the outdated version where my cost center was `other`. Likewise if you test with `firstCreation.Add(12 * time.Hour)`, that is a point between first and second, your query would return the outdated costcenter.

Reply:
Will add a test for this.